src/java/detectors/missing_check_on_method_output/MissingCheckOnMethodOutput.java

```
@@ -0,0 +1,49 @@
+package rules.missing_check_on_method_output;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import lombok.extern.log4j.Log4j;
+
+@Log4j
+class MissingCheckOnMethodOutput {
+
+    // {fact rule=missing-check-on-method-output@v1.0 defects=1}
+    private void writeMessageNonCompliant(String dirName, String fileName, String message) {
+        try {
+            File dir = new File(dirName);
+            if (!dir.exists()) {
+                // Noncompliant: code does not handle the case when mkdirs fails.
+                dir.mkdirs();
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)